### ISSUE TITLE
Handling TF softmax_cross_entropy_with_logits

### ISSUE THREAD

**Issue by eraly** [deeplearning4j/nd4j#2347]
Most TF graphs that run classification will run the softmax+crossentropy loss in one step with "tf.nn.softmax_cross_entropy_with_logits".  

When freezing the graph to run for inference we can give the freeze_graph script the node that precedes this which is the unnormalized log probabilities. However it would be cleaner and better as well easier to interpret if we had an option to add in a softmax at the very end. 

Maybe an override to the import function so that the user can specify to true? 

Something like -

TFGraphMapper.getInstance().importGraph(some_pb_stream,add_softmax_endnode=true)

**Comment by raver119** [comment]
That's actually question for @agibsonccc 
At this point we already have all TF loss functions implemented, except of sparse
